src/App.jsx

```diff
@@ -14,33 +20,95 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getPreparedGoods(goods, sortField, query) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+        case SORT_FIELD_ALPHABETICALLY:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (query) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, reversed);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="App">
+      <h1>
+        {' '}
+        {sortField}
+        {`${reversed ? ' reversed' : ''}`}
+      </h1>
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            onClick={() => setSortField(SORT_FIELD_ALPHABETICALLY)}
+            className={cn('button', 'is-info', {
+              'is-light': sortField !== SORT_FIELD_ALPHABETICALLY,
+            })}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            onClick={() => setSortField(SORT_FIELD_LENGTH)}
+            className={cn('button', 'is-success', {
+              'is-light': sortField !== SORT_FIELD_LENGTH,
+            })}
```

Comment:
Consider adding the `active` class to highlight the active button. You can do this by adding `'active': sortField === SORT_FIELD_LENGTH` to the className object.